### PR TITLE
Don't update shadow data if we created the entity locally

### PR DESCRIPTION
**Pull request by Vatyx**
#### Description
We were incorrectly resetting shadow data too early when there was state that was not sent.

#### Release note
Bugfix: Fixed state not getting replicated to SpatialOS when changing it when receiving authority

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp [hunk 1]
```diff
@@ -161,6 +162,15 @@ void USpatialActorChannel::UpdateShadowData()
 {
 	check(Actor);
 
+	// If this channel was responsible for creating the channel, we do not want to initialize our shadow data
+	// to the latest state since there could have been state that has changed between creation of the entity
+	// and gaining of authority. Revisit this with UNR-1034
+	// TODO: UNR-1029 - log when the shadow data differs from the current state of the Actor.
+	if (bCreatedNewEntity)
+	{
+		return;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: Set value to false.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 2]
```diff
@@ -268,6 +268,12 @@ void USpatialReceiver::HandleActorAuthority(Worker_AuthorityChangeOp& Op)
 				}
 				else if (Op.authority == WORKER_AUTHORITY_NOT_AUTHORITATIVE)
 				{
+					USpatialActorChannel* ActorChannel = NetDriver->GetActorChannelByEntityId(Op.entity_id);
+					if (ActorChannel != nullptr)
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: Move this into the if statement.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp (hunk 1 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: Oh I see - it's in authority loss. That's probably okay.

This time

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriver.h [hunk 3]
```diff
@@ -158,14 +158,14 @@ class SPATIALGDK_API USpatialNetDriver : public UIpNetDriver
 
 	void DelayedSendDeleteEntityRequest(Worker_EntityId EntityId, float Delay);
 
+	// Timer manager.
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriver.h]: Revert this

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialActorChannel.h [hunk 4]
```diff
@@ -132,6 +132,9 @@ class SPATIALGDK_API USpatialActorChannel : public UActorChannel
 	
 	void UpdateShadowData();
 
+	// If this actor channel is responsible for creating a new entity, this will be set to true once the entity is created.
+	bool bCreatedNewEntity;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialActorChannel.h]: Public variable? Please Sahil.

File: SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp [hunk 5]
```diff
@@ -204,7 +205,8 @@ void ComponentFactory::AddProperty(Schema_Object* Object, Schema_FieldId FieldId
 				}
 			}
 
-			if (NetGUID.IsValid())
+			// The secondary part of the check is only necessary until we have bulk reservation of entity ids
```
Comment (girayimprobable) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp]: Add a comment into the bulk reserve ticket. cc: @jnicholas-io

File: qq [hunk 6]
```diff
@@ -0,0 +1,122 @@
+[1mdiff --git a/SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp b/SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp[m
```
Comment (girayimprobable) [inline qq]: Wut?

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialActorChannel.h (hunk 4 above)
Comment (Vatyx) [reply SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialActorChannel.h]: I think I'm going to leave this since another function is just clutter

**Pull request merged**